scripts/iscsictl.py

```diff
@@ -0,0 +1,310 @@
+        result = str(self.ssh.cat('/proc/net/iet/volume'))
+        if 'iqn.2015-01.qa.cloud.suse.de:id01' not in result:
+            raise Exception('Unable to deploy the iSCSI target')
+
+
+class Initiator(ISCSI):
+    """Define and manage an iSCSI initiator node."""
+
+    # For now, we are going to use the discovery option for iSCSI to
+    # populate the database.  This simplify the deployment of a basic
+    # iSCSI scenario, with a single target point and multiple
+    # initiators.
+
+    def __init__(self, ssh, target_ip, mount):
+        """Initialize the Initiator instance with an ip and a mount point."""
+        super(Initiator, self).__init__(ssh)
+        self.target_ip = target_ip
+        self.mount = mount
+
+    def deploy(self, mkfs=False):
+        """Deploy, configure and persist an iSCSI initiator."""
+        self.zypper('open-iscsi')
+
+        # Default configuration only takes care of autentication
+        lines = (
+            'node.session.auth.authmethod = CHAP',
+            'node.session.auth.username = user',
+            'node.session.auth.password = passwd',
+            'discovery.sendtargets.auth.authmethod = CHAP',
+            'discovery.sendtargets.auth.username = user',
+            'discovery.sendtargets.auth.password = passwd',
+        )
+        for line in lines:
+            self.ssh.echo(line, '>> /etc/iscsid.conf')
+
+        # Persist and start the service
+        self.service('open-iscsi', True)
+        self.ssh.sh('rcopen-iscsi', 'restart')
+
+        # Discovery and login
+        result = self.ssh.iscsiadm('-m', 'discovery', '--type=st',
+                                   '--portal=%s' % self.target_ip)
+        name = 'iqn.2015-01.qa.cloud.suse.de:id01'
+        if name not in result:
+            raise Exception('Target %s not found' % name)
+        self.ssh.iscsiadm('-m', 'node', '-n', name, '--login')
+
+        if mkfs:
+            self.ssh.mkfs('/dev/sda1')
+
+        self.ssh.mkdir('-p', self.mount)
+        self.ssh.mount('/dev/sda1', self.mount)
+
+    def umount(self):
+        """Umount the shared device."""
+        self.ssh.umount(self.mount)
+
+        name = 'iqn.2015-01.qa.cloud.suse.de:id01'
+        self.ssh.iscsiadm('-m', 'node', '-n', name, '--logout')
+
+
+def test():
+    """Testing against mkcloud."""
+    # Adming node is going to be the target
+    admin = SSH('root', '192.168.124.10', 'linux')
+    # Initiators
+    node1 = SSH('root', '192.168.124.81', 'linux')
+
+    target = Target(admin)
+    target.deploy()
+
+    initiator1 = Initiator(node1, '192.168.124.10', '/mnt/iscsi')
+    initiator1.deploy(mkfs=True)
+
+    msg = 'this is a test from node1'
+    node1.echo(msg, '> /mnt/iscsi/test')
+    assert str(node1.cat('/mnt/iscsi/test').strip()) == msg, 'the file was not created on node1'
+    initiator1.umount()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Help iSCSI deployment for QA environments.')
```

Comment:
I think so (as I understand for the card), but I will remove this in hope to be more general.